### ISSUE TITLE
Set-content does not interpret the psdrive path correctly when the file does not exist

### ISSUE THREAD

**Issue by bingbing8** [PowerShell/PowerShell#2206]
## Steps to reproduce

```
1. md /home/yanbing/test                   
2. New-PSDrive -name TESTDRIVE -ROOT /home/bingbing/test -PSProvider FileSystem                                                                    

Name           Used (GB)     Free (GB) Provider      Root                                CurrentLocati
                                                                                                   on
----           ---------     --------- --------      ----                                -------------
TESTDRIVE          13.16        109.75 FileSystem    /home/bingbing/test                              


'abc' | Set-Content -NoNewline -Path TESTDRIVE:/testfile.txt                                                                                 
Set-Content : Could not find a part of the path 
'/home/yanbing/test/home/yanbing/test/home/yanbing/test/testfile.txt'.
At line:1 char:12
+ 'abc' | Set-Content -NoNewline -Path TESTDRIVE:testfile.txt
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   + CategoryInfo          : ObjectNotFound: (/home/yanbing/t...st/testfile.txt:String) [Set-Content 
  ], DirectoryNotFoundException
   + FullyQualifiedErrorId : GetContentWriterDirectoryNotFoundError,Microsoft.PowerShell.Commands.SetContentCommand
```
## Expected behavior

Should create a file and set the content
## Actual behavior

the path got appended three times so can not be found on the system.
If the file already exist, the content can be set to the right file.
## Environment data

```
Name                           Value                                                                  
----                           -----                                                                  
PSVersion                      6.0.0-alpha                                                            
PSEdition                      Core                                                                   
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                
BuildVersion                   3.0.0.0                                                                
GitCommitId                    v6.0.0-alpha.8                                                         
CLRVersion                                                                                            
WSManStackVersion              3.0                                                                    
PSRemotingProtocolVersion      2.3                                                                    
SerializationVersion           1.1.0.1    
```

**Comment by vors** [comment]
Confirmed on OS X.

@bingbing8 I updated the issue text to use proper formatting. Please, use markdown for issues. For details, see this comment https://github.com/PowerShell/PowerShell/issues/2144#issuecomment-243747540

**Comment by charub** [comment]
Issue repro's on PowerShell Core as well. More Info :

PS C:\Users\charub> New-PSDrive -Name x -Root \\scratch2\scratch\charub -PSProvider FileSystem

Name           Used (GB)     Free (GB) Provider      Root
----           ---------     --------- --------      ----
x                                      FileSystem    \\scratch2\scratch\charub


PS C:\Users\charub> 'abc' | Set-Content -path x:\testfile.txt -NoNewline
Set-Content : Could not find a part of the path '\\scratch2\scratch\charub\scratch2\scratch\charub\scratch2\scratch\charub\testfile.txt'.
At line:1 char:9
+ 'abc' | Set-Content -path x:\testfile.txt -NoNewline
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (\\scratch2\scra...ub\testfile.txt:String) [Set-Content], DirectoryNotFoundException
    + FullyQualifiedErrorId : GetContentWriterDirectoryNotFoundError,Microsoft.PowerShell.Commands.SetContentCommand

PS C:\Users\charub> $PSVersionTable

Name                           Value
----                           -----
PSRemotingProtocolVersion      2.3
CLRVersion
WSManStackVersion              3.0
SerializationVersion           1.1.0.1
BuildVersion                   3.0.0.0
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSEdition                      Core
PSVersion                      6.0.0-alpha
GitCommitId                    v6.0.0-alpha.10

**Comment by daxian-dbw** [comment]
Resolved by PR #2392
